### PR TITLE
Allow to use task retries without until block

### PR DESCRIPTION
**Pull request by barlik**
##### SUMMARY

This change will make task retries work without having an `until` block. This is useful in situations when you are not interested in the output value and/or you simply want to retry on a failure. This allows you to omit `register` and `until` parameters and makes the task easier to read.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
core engine

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (task-retries 7dda294680) last updated 2018/07/22 11:26:54 (GMT +100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/kerny/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kerny/dev/github/ansible/lib/ansible
  executable location = /home/kerny/dev/github/ansible/bin/ansible
  python version = 3.6.6 (default, Jun 27 2018, 13:11:40) [GCC 8.1.1 20180531]
```

##### ADDITIONAL INFORMATION

The change will allow you to simplify:
```
    - name: Retry URL call up to five times until we get 200 status code
      uri:
         url: https://httpbin.org/status/200
      register: out
      until: out is successful
      retries: 5
```
into:
```
    - name: Retry URL call up to five times until we get 200 status code
      uri:
         url: https://httpbin.org/status/200
      retries: 5
```

This PR also fixes a small bug (inconsistency) whereby having `retries: ` without a value defaults to 2 retries instead of 3.

### CODE REVIEW

**Comment by barlik** [comment]
Proposal discussion at: https://github.com/ansible/proposals/issues/132

**Review by achinthagunasekara — APPROVED** [review]

**Comment by barlik** [comment]
Any update on this?

**Comment by samccann** [comment]
@dagwieers @bcoca  - any update here? looks from the proposal that the idea has merit but haven't seen any action on code reviews here in the PR?

**Comment by lukatera** [comment]
@barlik can you resolve conflicts?

**Comment by chrisjohnson00** [comment]
Bump

**Comment by barlik** [comment]
I've rebased the code and fixed merge conflicts. It's been a **very long** time since I've created this PR so hopefully everything still works.

**Comment by ciaby** [comment]
Is this PR still open? Would be very handy to have...

**Comment by caleb15** [comment]
@bcoca can you give this a review please? If you're busy is there a different person who is available to review this?

**Comment by bcoca** [comment]
@caleb15 this is not blocked on me, anyone can review, it is on my list .. but i have a long list

**Comment by caleb15** [comment]
Thanks for the fast response. @mattclay @samccann are one of you available to review? I'd be happy to donate a coffee to you/charity in return.

**Comment by samccann** [comment]
@caleb15 alas, I can't review python changes. The docs portion LGTM.
The core team meets at 3pm ET on Tuesdays and 11AM ET on Thursdays on #ansible-meeting on IRC. You could try to bring it up there and see if that gets any review/feedback.

**Comment by bcoca** [comment]
as noted in meeting, we need more tests for current behaviour before we can proceed.
https://github.com/ansible/community/issues/543#issuecomment-662073579

also some had issues with the current implementation, will leave up to them to add specific comments.

**Comment by bcoca** [comment]
closing in favor of #76101

**Pull request closed without merging**
